src/bindings/c/include/openvino/c/ov_core.h

```diff
@@ -0,0 +1,268 @@
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_read_model(const ov_core_t* core, const char* model_path, const char* bin_path, ov_model_t** model);
+
+/**
+ * @brief Reads models from IR/ONNX/PDPD formats.
+ * @ingroup Core
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_str String with a model in IR/ONNX/PDPD format.
+ * @param weights Shared pointer to a constant tensor with weights.
+ * @param model A pointer to the newly created model.
+ * Reading ONNX/PDPD models does not support loading weights from the @p weights tensors.
+ * @note Created model object shares the weights with the @p weights object.
+ * Thus, do not create @p weights on temporary data that can be freed later, since the model
+ * constant data will point to an invalid memory.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_read_model_from_memory(const ov_core_t* core,
+                               const char* model_str,
+                               const ov_tensor_t* weights,
+                               ov_model_t** model);
+
+/**
+ * @brief Creates a compiled model from a source model object.
+ * Users can create as many compiled models as they need and use
+ * them simultaneously (up to the limitation of the hardware resources).
+ * @ingroup Core
+ * @param core A pointer to the ie_core_t instance.
+ * @param model Model object acquired from Core::read_model.
+ * @param device_name Name of a device to load a model to.
+ * @param compiled_model A pointer to the newly created compiled_model.
+ * @param property Optional pack of pairs: (property name, property value) relevant only for this load operation
+ * operation.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_compile_model(const ov_core_t* core,
+                      const ov_model_t* model,
+                      const char* device_name,
+                      ov_compiled_model_t** compiled_model,
+                      const ov_property_t* property);
+
+/**
+ * @brief Reads a model and creates a compiled model from the IR/ONNX/PDPD file.
+ * This can be more efficient than using the ov_core_read_model_from_XXX + ov_core_compile_model flow,
+ * especially for cases when caching is enabled and a cached model is available.
+ * @ingroup Core
+ * @param core A pointer to the ie_core_t instance.
+ * @param model_path Path to a model.
+ * @param device_name Name of a device to load a model to.
+ * @param compiled_model A pointer to the newly created compiled_model.
+ * @param property Optional pack of pairs: (property name, property value) relevant only for this load operation
+ * operation.
+ * @return Status code of the operation: OK(0) for success.
+ */
+OPENVINO_C_API(ov_status_e)
+ov_core_compile_model_from_file(const ov_core_t* core,
+                                const char* model_path,
```

Comment:
is it the same as usual ov_core_read_model, but w/o bin path? can we merge these methods then? we can pass bin=null I suppose

Reply:
It  is for compile model, and ov_core_read_model is for read model from IR files.